test/run-k8s-integration-ci.sh

```diff
@@ -93,11 +93,18 @@ else
 fi
 
 if [ "$deployment_strategy" = "gke" ]; then
-  if [ "$gke_release_channel" ]; then
+  # To leverage old cluster versions, we need set both the release channel and
+  # the cluster version.
+  if [ "$gke_release_channel" = "extended" ]; then
```

Comment:
Does this work if $gke_release_channel is empty? I'm always a little confused by bash parsing, but if the string is empty you may end up with if [ = extended ] which is a parse error

Reply:
Good catch, I did a little more looking and I believe I've got a better version now.

That first part is right (gke_release_channel is defaulted to empty string in its var declaration).  Stackoverflow says `[ thing = otherThing ]` is [the right POSIX solution](https://superuser.com/a/1076805).

The second one was more ambiguous.  I've changed it to check for a non-empty variable:

```
  elif [ -n "$gke_release_channel" ]; then
  ```

Reply:
ah, the quotes matter. [ $unknown = foo ] fails to parse, but [ "$unknown" = foo ] is okay. Interesting.

+1 to using -n, it's a little more clear (although [ ] seems to be valid anyway)